homeassistant/components/switch/hydrawise.py

```diff
@@ -0,0 +1,119 @@
+    hydrawise = hass.data[DATA_HYDRAWISE].data
+
+    default_watering_timer = config.get(CONF_WATERING_TIME)
+
+    sensors = []
+    for sensor_type in config.get(CONF_MONITORED_CONDITIONS):
+        # create a switch for each zone
+        for zone in hydrawise.relays:
+            sensors.append(
+                HydrawiseSwitch(default_watering_timer,
+                                zone,
+                                sensor_type))
+
+    add_devices(sensors, True)
+    return True
+
+
+class HydrawiseSwitch(HydrawiseEntity, SwitchDevice):
+    """A switch implementation for Hydrawise device."""
+
+    def __init__(self, default_watering_timer, *args):
+        """Initialize a switch for Hydrawise device."""
+        super().__init__(*args)
+        self._default_watering_timer = default_watering_timer
+
+    @property
+    def is_on(self):
+        """Return true if device is on."""
+        return self._state
+
+    def turn_on(self, **kwargs):
+        """Turn the device on."""
+        if self._sensor_type == 'manual_watering':
+            self.hass.data['hydrawise'].data.run_zone(
+                self._default_watering_timer, (self.data.get('relay')-1))
+        elif self._sensor_type == 'auto_watering':
+            self.hass.data['hydrawise'].data.suspend_zone(
+                0, (self.data.get('relay')-1))
+        self._state = True
+
+    def turn_off(self, **kwargs):
+        """Turn the device off."""
+        if self._sensor_type == 'manual_watering':
+            self.hass.data['hydrawise'].data.run_zone(
+                0, (self.data.get('relay')-1))
+        elif self._sensor_type == 'auto_watering':
+            self.hass.data['hydrawise'].data.suspend_zone(
+                365, (self.data.get('relay')-1))
+        self._state = False
+
+    def update(self):
+        """Update device state."""
+        mydata = self.hass.data['hydrawise'].data
+        _LOGGER.debug("Updating Hydrawise switch: %s", self._name)
+        if self._sensor_type == 'manual_watering':
+
+            if mydata.running is None or not mydata.running:
+                self._state = False
+            else:
+                self._state = int(
+                    mydata.running[0]['relay']) == self.data.get('relay')
+        elif self._sensor_type == 'auto_watering':
+            for relay in mydata.relays:
+                if relay.get('relay') == self.data.get('relay'):
+                    if relay.get('suspended') is not None:
+                        self._state = False
+                    else:
+                        self._state = True
+                    break
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        temp = self.hass.data['hydrawise'].data.\
+            controller_info['controllers'][0]['last_contact_readable']
+        return {
+            ATTR_ATTRIBUTION: CONF_ATTRIBUTION,
+            "identifier": self.data.get('relay'),
+            "last contact": temp
```

Comment:
Use snakecase lowercase strings for attributes.

Reply:
Fixed.